plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BooleanWriteFunction.java

```diff
@@ -27,4 +29,30 @@ default Class<?> getJavaType()
 
     void set(PreparedStatement statement, int index, boolean value)
             throws SQLException;
+
+    static BooleanWriteFunction of(int nullJdbcType, BooleanWriteFunctionImplementation implementation)
+    {
+        requireNonNull(implementation, "implementation is null");
+
+        return new BooleanWriteFunction() {
+            @Override
+            public void set(PreparedStatement statement, int index, boolean value) throws SQLException
+            {
+                implementation.set(statement, index, value);
+            }
+
+            @Override
+            public void setNull(PreparedStatement statement, int index) throws SQLException
+            {
+                statement.setNull(index, nullJdbcType);
+            }
+        };
+    }
+
+    @FunctionalInterface
```

Comment:
I thought this suggestion would reduce the amount of boilerplate (and it does, in `StandardColumnMappings`, but not here).
I wonder if these "XWriteFunctionImplementation" interfaces are redundant? If we just reuse `XWriteFunction` as the FunctionalInterface 
```
@FunctionalInterface
public interface DoubleWriteFunction
        extends WriteFunction
{
    @Override
    default Class<?> getJavaType()
    {
        return double.class;
    }

    void set(PreparedStatement statement, int index, double value)
            throws SQLException;

    static DoubleWriteFunction of(int nullJdbcType, DoubleWriteFunction implementation)
    {
        return new DoubleWriteFunction() {
            @Override
            public void set(PreparedStatement statement, int index, double value) throws SQLException
            {
                implementation.set(statement, index, value);
            }

            @Override
            public void setNull(PreparedStatement statement, int index) throws SQLException
            {
                statement.setNull(index, nullJdbcType);
            }
        };
    }
}
```
(Or is there an issue with marking the `BooleanWriteFunction` as a FunctionalInterface, in that if we add non-default methods in future we would have to revisit this?)

Reply:
I'd prefer to keep it consistent with existing `ObjectWriteFunction` shape.